java-checks/src/main/java/org/sonar/java/filters/SpringFilter.java

```diff
@@ -0,0 +1,139 @@
+      /* S1185 */ MethodOnlyCallsSuperCheck.class,
+      /* S1258 */ AtLeastOneConstructorCheck.class,
+      /* S2226 */ ServletInstanceFieldCheck.class);
+  }
+
+  @Override
+  public void visitClass(ClassTree tree) {
+    excludeLinesIfTrue(isTransactional(tree), tree, MethodOnlyCallsSuperCheck.class);
+    excludeLinesIfTrue(hasAutowiredField(tree), tree.simpleName(), AtLeastOneConstructorCheck.class);
+    super.visitClass(tree);
+  }
+
+  @Override
+  public void visitVariable(VariableTree tree) {
+    excludeLinesIfTrue(isAutowired(tree), tree, ServletInstanceFieldCheck.class);
+    super.visitVariable(tree);
+  }
+
+  @Override
+  public void visitMethod(MethodTree tree) {
+    MethodSymbol symbol = tree.symbol();
+    if (tree.is(Tree.Kind.CONSTRUCTOR)) {
+      SymbolMetadata ownerMetadata = symbol.owner().metadata();
+      excludeLinesIfTrue(S107_CLASS_ANNOTATION_EXCEPTIONS.stream().anyMatch(ownerMetadata::isAnnotatedWith), tree.simpleName(), TooManyParametersCheck.class);
+    } else {
+      SymbolMetadata methodMetadata = symbol.metadata();
+      excludeLinesIfTrue(S107_METHOD_ANNOTATION_EXCEPTIONS.stream().anyMatch(methodMetadata::isAnnotatedWith), tree.simpleName(), TooManyParametersCheck.class);
+    }
+    super.visitMethod(tree);
+  }
+
+  private static boolean hasAutowiredField(ClassTree tree) {
+    return tree.members().stream()
+      .filter(member -> member.is(Tree.Kind.VARIABLE))
+      .map(VariableTree.class::cast)
+      .anyMatch(SpringFilter::isAutowired);
+  }
+
+  private static boolean isAutowired(VariableTree tree) {
+    return tree.symbol().metadata().isAnnotatedWith(AUTOWIRED)
+      // missing semantic
+      || hasUnknownAnnotationWithName(tree.modifiers(), "Autowired");
+  }
+
+  private static boolean isTransactional(ClassTree tree) {
+    return tree.symbol().metadata().isAnnotatedWith("org.springframework.transaction.annotation.Transactional")
+      // missing semantic
+      || hasUnknownAnnotationWithName(tree.modifiers(), "Transactional");
+  }
+
+  private static boolean hasUnknownAnnotationWithName(ModifiersTree modifiers, String annotation) {
+    // Token based check in case of missing semantic
+    List<AnnotationTree> annotations = modifiers.annotations();
+    if (annotations.isEmpty()) {
+      return false;
+    }
+    return annotations.stream()
+      .map(AnnotationTree::annotationType)
+      .filter(a -> a.symbolType().isUnknown())
+      // unknown annotation potentially matching - checking for tokens
+      .filter(tree -> tree.is(Tree.Kind.IDENTIFIER, Tree.Kind.MEMBER_SELECT))
+      .map(ExpressionTree.class::cast)
+      .map(ExpressionsHelper::concatenate)
+      .anyMatch(annotationName -> annotationName.endsWith(annotation));
+  }
+
+  private void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck> rule) {
+    if (shouldExclude) {
+      excludeLines(tree, rule);
+    } else {
+      acceptLines(tree, rule);
+    }
+  }
```

Comment:
I noticed that `LombokFilter` has the same method, plus one that takes more than one `JavaCheck` as varargs.
I suggest we move both methods in `LombokFilter` inside `BaseTreeVisitorIssueFilter`, as they can be shared between `SpringFilter` and `LombokFilter` for now, and could also benefit future filters.

So I would remove these two methods from `LombokFilter`:
```
  @SafeVarargs
  private final void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck>... rules) {
    for (Class<? extends JavaCheck> rule : rules) {
      excludeLinesIfTrue(shouldExclude, tree, rule);
    }
  }

  private void excludeLinesIfTrue(boolean shouldExclude, Tree tree, Class<? extends JavaCheck> rule) {
    if (shouldExclude) {
      excludeLines(tree, rule);
    } else {
      acceptLines(tree, rule);
    }
  }
```
move them inside `BaseTreeVisitorIssueFilter` and change their visibility to `protected`. Then also remove the `excludeLinesIfTrue` from `SpringFilter`

Reply:
good point.